### ISSUE TITLE
[Feature Request] Self Hosted Notifiarr

### ISSUE THREAD

**Issue by DrSeussFreak** [Notifiarr/website#19]
I understand that this would be down the road, but to be able to host this myself would be an awesome feature :)

---------------------
IMPORTANT EDIT: 
I am going to hijack this initial post so everyone can see a response without having to scroll a bunch. There are no client side API keys stored anywhere outside of local config files, so that isn't an issue to be concerned with. If using the client then every request is validated with the Notifiarr API key against one stored locally and if they dont match, the request is also thrown out. There are no delete endpoints setup as well. If not using the client, it is no different than using a Discord connect in terms of being a "3rd party service". There is nothing stored on the site that can hurt a remote setup.

Some current things that come to mind are keys that you cant even make anymore are needed, paid services are used, a cache of data (IMDb, TMDb, TVDb, Skyhook, MB, GR, OMDB, etc), setting up an actual bot with discord and not the webhook url bot (required for editing posts, reactions, etc), setting up a build process and fixing things + pushing them without having users wait weeks or more for the smallest changes, etc

I personally have no issue with those who dont want to use it, i am OK with that :) so for anyone else commenting, please dont compare to self hosted alternatives as 1. i dont care as that is their decision (just being straight forward and not meant to be rude) and dont use them & 2. there are no self hosted alternatives that do everything this does and will be extended to do in the near future

To each their own, you're welcome to add a note for supporting the idea but just keep all this in mind.

-- nitsua
----------------

**Comment by LongDono** [comment]
Agreed. I am not comfortable adding in a third party for these kind of notifications.

**Comment by bakerboy448** [comment]
Just for explicit clarity: it will not be made self hosted anytime soon

too many APIs, cache's, and other moving parts needed

**Comment by davidnewhall** [comment]
As a dev with a small stake in the success of this product, I don't believe it can or ever will be self-hostable. I still think one day the source code, or most of it, can be open sourced. That will allow others to help find and fix bugs. There is unlikely to ever be a point at which any of those involved with this product (like Rox, trash, myself, nitsua and bakerboy) provide support for running it _yourself_. As nitsua pointed out, there are certain things he's built data stores around that are simply no longer available. It may not be in his best interest to expose or provide that data. This is what gives his product a competitive advantage and it's a large part of what makes this product better than any competitor.

**Comment by neonvoidx** [comment]
> As a dev with a small stake in the success of this product, I don't believe it can or ever will be self-hostable. I still think one day the source code, or most of it, can be open sourced. That will allow others to help find and fix bugs. There is unlikely to ever be a point at which any of those involved with this product (like Rox, trash, myself, nitsua and bakerboy) provide support for running it _yourself_. As nitsua pointed out, there are certain things he's built data stores around that are simply no longer available. It may not be in his best interest to expose or provide that data. This is what gives his product a competitive advantage and it's a large part of what makes this product better than any competitor.

That's fair however take into consideration that someone is going to come along very soon and essentially make a clone of this that can be self hosted, and lets be honest probably better, i.e a progressive web app. Today is a prime example, notifiarr is running very slow today having issues and I'm not even getting notifies via discord webhooks

**Comment by austinwbest** [comment]
Well, if someone wants to spend that time it is on them to do it and we will wish them well. If it is running slow it can be brought up in discord and i can look at what is going on, same for if notifications dont send. However this has nothing to do with discord webhooks as it uses the API directly so im not really sure how to address that one.

In the past few years there have been all of a handful of people who have mentioned self hosted being a priority it and considering there have been over 8k member join since this ticket was raised i think its ok. Cant please everyone but i can do my best to add all the options & settings that users request and the others can hop over to Apprise (or Discord webhooks for the apps that support it) with no hard feelings.

The site is currently going through an entire rework for the template to get it up to Bootstrap5 which will mitigate many of the mobile user complaints.

There is a 48 core, 400GB memory server being migrated for Notifiarr to run on as well (with a mirror server & a 3rd 24 core for shared storage) which will also mitigate the resources being higher than i would like.

So lets be honest, if you dont like it dont use it. No harm no foul.

**Comment by neonvoidx** [comment]
Fair enough
